packages/integrations/markdoc/README.md

```diff
@@ -0,0 +1,342 @@
+
+__`astro.config.mjs`__
+
+```js ins={2} "markdoc()"
+import { defineConfig } from 'astro/config';
+import markdoc from '@astrojs/markdoc';
+
+export default defineConfig({
+  // ...
+  integrations: [markdoc()],
+});
+```
+
+
+### Editor Integration
+
+[VS Code](https://code.visualstudio.com/) supports Markdown by default. However, for Markdoc editor support, you may wish to add the following setting in your VSCode config. This ensures authoring Markdoc files provides a Markdown-like editor experience.
+
+```json title=".vscode/settings.json"
+"files.associations": {
+    "*.mdoc": "markdown"
+}
+```
+
+## Usage
+
+Markdoc files can only be used within content collections. Add entries to any content collection using the `.mdoc` extension:
+
+```sh
+src/content/docs/
+  why-markdoc.mdoc
+  quick-start.mdoc
+```
+
+Then, query your collection using the [Content Collection APIs](https://docs.astro.build/en/guides/content-collections/#querying-collections):
+
+```astro
+---
+import { getEntryBySlug } from 'astro:content';
+
+const entry = await getEntryBySlug('docs', 'why-markdoc');
+const { Content } = await entry.render();
+---
+
+<!--Access frontmatter properties with `data`-->
+<h1>{entry.data.title}</h1>
+<!--Render Markdoc contents with the Content component-->
+<Content />
+```
+
+📚 See the [Astro Content Collection docs][astro-content-collections] for more information.
+
+## Configuration
+
+`@astrojs/markdoc` offers configuration options to use all of Markdoc's features and connect UI components to your content.
+
+### Using components
+
+You can add Astro and UI framework components (React, Vue, Svelte, etc.) to your Markdoc using both [Markdoc tags][markdoc-tags] and HTML element [nodes][markdoc-nodes].
+
+#### Render Markdoc tags as Astro components
+
+You may configure [Markdoc tags][markdoc-tags] that map to components. You can configure a new tag from your `astro.config` using the `tags` attribute.
+
+
+```js
+// astro.config.mjs
+import { defineConfig } from 'astro/config';
+import markdoc from '@astrojs/markdoc';
+
+// https://astro.build/config
+export default defineConfig({
+	integrations: [
+		markdoc({
+			tags: {
+				aside: {
+					render: 'Aside',
+          attributes: {
+            // Component props as attribute definitions
+            // See Markdoc's documentation on defining attributes
+            // https://markdoc.dev/docs/attributes#defining-attributes
+            type: { type: String },
+          }
+				},
+			},
+		}),
+	],
+});
+```
+
+Then, you can wire this render name (`'Aside'`) to a component from the `components` prop via the `<Content />` component:
+
+
+```astro
+---
+import { getEntryBySlug } from 'astro:content';
+import Aside from '../components/Aside.astro';
+
+const entry = await getEntryBySlug('docs', 'why-markdoc');
+const { Content } = await entry.render();
+---
+
+<Content
+  components={{
+    Aside: Aside,
```

Comment:
Because these are exactly the same name, and both capitalized, I have to admit I'm not sure which `Aside` is which. I know they're "wired" together, but it's not obvious which one is "defining" the other.

a) They don't *have* to have the same name, right? We can call our component `CustomAside` for example? (Which I think is cleaner than trying to write more words to describe the explicit relationship)

b) I think also seeing an example of this being used in a Markdoc file is useful here. What we can see is that we're rendering "all our Markdoc content, with a custom aside figured". I think then seeing a Markdoc snippet is helpful. In their own docs, their tags seem to be lower case (e.g. the`callout` example). But we have two upper case `Asides`, so I might be confused about using a tag.. do I do lower case becase that seems to be how you write Markdoc? Is it upper case because that's how we've wired it here?

Reply:
Discussed offline. Changing to the shortened JS object syntax (`components={{ Aside }}`) to follow best practices and assume an intermediate programmer knowledge